### ISSUE TITLE
Deploy with kamal 

### ISSUE THREAD

**Issue by gedw99** [leomorpho/goship#3]
https://github.com/MightyMoud/sidekick?tab=readme-ov-file#inspiration

Would be awesome if we used this . Being golang it’s easier to manage. It’s still a little new. It works on DO and Hetzner because they both provide nested virtualisation . Just like Kamal I believe .

curious what you think .

I think we could also use gossip to control sidekick for managing fleets . Just an idea ..

**Comment by leomorpho** [comment]
Hey thanks for this idea! I must say I really like it, first time I hear of this project.

I am not quite clear on whether Sidekick can support deploying more than a single binary in a docker file? For the current state of Goship, it still depends on Redis. Now, it's not that hard to remove it, and asynq (which requires redis) can be switched with [backlite](https://github.com/mikestefanello/backlite) like in [this PR](https://github.com/mikestefanello/pagoda/pull/72/files) from [Pagoda](https://github.com/mikestefanello/pagoda?tab=readme-ov-file#tasks). Personally, I'm just not 100% convinced by backlite yet...and I've been looking at other task runners but haven't found one I'm in love with.

It seems reasonable to add different ways to deploy, though we'd need to keep that clean. Currently, kamal files live in the root, and adding another deployment tool might add more stuff? If that's the case, we'd need to revisit the file organization just to keep it sane. Long-term, if Sidekick turns out to be better and easier to manage, it might make sense to drop Kamal for a simpler tool, and one that is in Go too.

Would you be interested to help integrating Sidekick? Let me know :)

**Comment by leomorpho** [comment]
Sidekick support sounds good to me. For the docker-compose problem, I think we could ignore it to start and simply deploy for a single binary? Redis is the last standalone service dependency, and apart from the task runner, it's only used for SSE events, a feature that can be reimplemented to remove that need for Redis. Thoughts?

I've read up a bit about nats, it's new to me. One concern there for me is that it increases the complexity. I also would want to eventually keep this as a single binary, and only have the option for Postgres/Redis if one requires it.

**Comment by leomorpho** [comment]
Currently to run my Procfile I use `overmind`, but that's used only for local dev. I don't think there is any need for docker-compose as the goal is to remove Redis as a dependency, so the final project is a simple binary that can be deployed with whatever tool suits one's need. 

I am totally in to add more deployment ways, but it needs to remain highly organized, i.e. all deploy-specific configs/code should live within for example a `deploy` directory, with one child directory per deployment method. I think the project can already feel overwhelming when first picked up by someone, and I want to go towards a direction of simplicity and functional organization. As long as whatever is added fits that paradigm, I am all for it.

NATs seems pretty cool to me, but I am not convinced increasing the system complexity with it is warranted. I took a look at [asyncjobs](https://github.com/choria-io/asyncjobs) and it seems interesting. I couldn't find any easy to use UI for the task monitoring. That is actually something I'd want. It would either need to be a standalone tool (like asynqmon) or built into goship, which frankly seems out of scope. The reason I'd want a UI is it makes one's life simpler for monitoring in prod.

I think a good first step, if you plan to use `sidekick` with goship is to add support for it? Let me know what you think. That would also keep the scope limited for now. What do you think?

**Comment by leomorpho** [comment]
Hey, I had started upgrading to rely solely on SQLite though I had not finished, not sure when I will get to it. I definitely would keep Kamal, really like that tool.

**Comment by leomorpho** [comment]
Yes, I mean this [kamal](https://kamal-deploy.org). Closing because not planning to support anything else than Kamal. It's easy enough to integrate any deployment tool you want though.
